include/envoy/ssl/certificate_validation_context_config.h

```diff
@@ -40,6 +41,12 @@ class CertificateValidationContextConfig {
    */
   virtual const std::vector<std::string>& verifySubjectAltNameList() const PURE;
 
+  /**
+   * @return The subject alt name matchers to be verified, if enabled. Otherwise, ""
+   */
+  virtual const std::vector<::envoy::type::matcher::StringMatcher>&
+  matchSubjectAltNameList() const PURE;
```

Comment:
nit: since you are editing this code, I would probably call this `subjectAltNameMatchers()` or something like that?